File: packages/runtime-java/src/main/java/besdk/runtime/ApiKeyAuth.java
package besdk.runtime;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/**
 * An API key, in a header or the query string.
 *
 * <p>The query variant exists because specs declare it, not because it is a good idea — a key in a
 * URL lands in access logs and browser history. besdk honours what the spec says and does not
 * editorialise.
 */
public final class ApiKeyAuth implements Auth {

  private final String key;
  private final String name;
  private final boolean inQuery;

  public ApiKeyAuth(String key, String name, boolean inQuery) {
    this.key = key;
    this.name = name;
    this.inQuery = inQuery;
  }

  @Override
  public Applied apply(Map<String, String> headers, Map<String, List<String>> query) {
    if (inQuery) {
      Map<String, List<String>> out = new LinkedHashMap<>(query);
      out.put(name, List.of(key));
      return new Applied(headers, out);
    }
    Map<String, String> out = new LinkedHashMap<>(headers);
    out.put(name, key);
    return new Applied(out, query);
  }
}

File: packages/runtime-php/src/ApiKeyAuth.php
<?php

declare(strict_types=1);

namespace Graft\Runtime;

/**
 * An API key, in a header or the query string.
 *
 * The query variant exists because specs declare it, not because it is a good idea — a key in a URL lands
 * in access logs and browser history. graft honours what the spec says and does not editorialise.
 */
final class ApiKeyAuth implements Auth
{
    public function __construct(
        private readonly string $key,
        private readonly string $name,
        private readonly bool $inQuery = false,
    ) {}

    public function apply(array $headers, array $query): array
    {
        if ($this->inQuery) {
            $query[$this->name] = [$this->key];
        } else {
            $headers[$this->name] = $this->key;
        }

        return [$headers, $query];
    }
}
